back/app/controllers/web_api/v1/users_controller.rb

```diff
@@ -18,8 +18,14 @@ def index
     @users = @users.blocked if params[:only_blocked]
     @users = @users.search_by_all(params[:search]) if params[:search].present?
 
+    private_attributes_in_export = AppConfiguration.instance.settings['core']['private_attributes_in_export']
     # Filter by project participants
     if params[:project].present?
+      # Block access to project participants list if private_attributes_in_export is disabled
+      if !private_attributes_in_export
+        raise ActiveRecord::RecordNotFound
+      end
+
```

Comment:
Checked out your code and was seeing odd behaviour. The realised that the logic needs to be explicitly checking for `!= false` as the setting is undefined by default, but that means true as you have to explicitly turn it off in adminHQ for it to write a value!! You might want to check this works though...

Small suggestion too on style and also a good idea to keep variables scoped only to the block (in this case `if params...`) that they are used.

```suggestion
   
    # Filter by project participants
    if params[:project].present?
      # Block access to project participants list if private_attributes_in_export is disabled
      private_attributes_in_export = AppConfiguration.instance.settings['core']['private_attributes_in_export'] != false
      raise ActiveRecord::RecordNotFound unless private_attributes_in_export

```